src/ipa_pc/mod.rs

```diff
@@ -1023,19 +1094,26 @@ impl<G: AffineCurve, D: Digest> PolynomialCommitment<G::ScalarField> for InnerPr
 
         let key_len = ck.comm_key.len();
         let p_len = batch_polynomial.coeffs.len();
+        let segments_count = std::cmp::max(1, p_len / key_len + if p_len % key_len != 0 { 1 } else { 0 });
+
+        let batch_randomness = if has_hiding {
+            Randomness::rand(segments_count, false, &mut rand::thread_rng())
+        } else {
+            Randomness::empty(segments_count)
+        };
 
         // Commitment of the h(X) polynomial
         let batch_commitment: Vec<G>;
 
         if p_len > key_len {
 
-            batch_commitment = (0..p_len / key_len + if p_len % key_len != 0 { 1 } else { 0 }).into_par_iter().map(
+            batch_commitment = (0..segments_count).into_iter().map(
```

Comment:
Doesn't treat the `cm_commit()` function segmentation? Why calling the cm_commit() on each of the segment commitments separately?

Reply:
No, cm_commit function doesn't work with segmentation. We have higher level function "commit" which splits polys into segments, and calls cm_commit for each segment. "cm_commit" is lower level function which expects that parameters are already segmented.